ttnctl/cmd/clients_edit.go

```diff
@@ -25,56 +24,48 @@ var clientEditCmd = &cobra.Command{
 
 		account := util.GetAccount(ctx)
 
-		clientCtx := ctx.WithField("OAuthClientID", name)
+		ctx = ctx.WithField("OAuthClientName", name)
 
-		// Verifying that there is information to edit
-		var flagsSet bool
-		cmd.Flags().Visit(func(*pflag.Flag) { flagsSet = true })
-		if !flagsSet {
-			clientCtx.Fatal("No information to edit")
-		}
-
-		clientCtx.Info("Retrieving OAuth client...")
+		ctx.Info("Retrieving OAuth client...")
 		client, err := account.FindOAuthClient(name)
 		if err != nil {
-			clientCtx.WithError(err).Fatal("Could not find OAuth client")
+			ctx.WithError(err).Fatal("Could not retrieve OAuth client")
 		}
 		if client == nil {
-			clientCtx.Fatal("No OAuth client returned by the server")
-		}
-		clientCtx.Info("Retrieved OAuth client")
-
-		if newName, err := cmd.Flags().GetString("name"); err != nil {
-			clientCtx.WithError(err).Fatal("Couldn't parse new name")
-		} else if newName != "" {
-			client.Name = newName
-			clientCtx = ctx.WithField("OAuthClientID", newName)
+			ctx.Fatal("No OAuth client returned by the server")
 		}
+		ctx.Debug("Retrieved OAuth client")
 
+		var edited bool
 		if newDescription, err := cmd.Flags().GetString("description"); err != nil {
-			clientCtx.WithError(err).Fatal("Couldn't parse description")
+			ctx.WithError(err).Fatal("Nothing to parse")
```

Comment:
This error occurs when there is a runtime issue with the CLI flags (such as *using a viper flag that wasn't pre-declared*). It doesn't happen because the user didn't insert a description (the program returns an error before that), or because there was no argument given - these are cases handled by viper. So `err` is returned only if there is an implementation error.